collectd/collectd.go

```diff
@@ -107,7 +107,11 @@ func (s *Server) serve() {
 			continue
 		}
 		if n > 0 {
-			s.handleMessage(buffer[:n])
+			var messageBuffer = make([]byte, len(buffer))
+			copy(messageBuffer, buffer)
+
+			// Make packet processing in a separate goroutine so it prevents UDP packet loss
```

Comment:
Thanks.

We'll need to think about this as it means a go-routine per UDP package. I don't think that is an issue, but we'll need to check. We may also need to extend our use of wait groups, so that the collectd module does not close until all these goroutines stop. Right now the code only waits for the main "listen" goroutine to stop.